### ISSUE TITLE
[3.7] [regression] Article Blog layout

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#12519]
### Steps to reproduce the issue
1. Set article options for blog layout to 1 column and confirm that a menu item set to use the globals (ie no numbers entered in the column field) and check that the blog layout is in one column

2. Set article options for blog layout to 2 columns and confirm that a menu item set to use the globals (ie no numbers entered in the column field) and check that the blog layout is in two columns

3. Set the blog layout option in the menu item  to 1 column

### Expected result
After step 3 it should be in 1 column

### Actual result
It is in 2 columns

### Additional comment
Basically none of the blog layout settings at the menu level are overriding the settings at the component level

**Comment by brianteeman** [comment]
I ran out of battery but I suspect the same is true for all options at the menu level<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12519">issues.joomla.org/tracker/joomla-cms/12519</a>.</sub>

**Comment by ghazal** [comment]
> Basically none of the blog layout settings at the menu level are overriding the settings at the component level

Confirmed.

**Comment by Bakual** [comment]
It's actually not only the blog layout, it's everytime we merge component parameters with menu parameters, probably anytime we merge registry objects using `registry_object->merge($another-registry-object)`.
I vaguely remember we had this issue once before.

**Comment by Bakual** [comment]
@mbabker Do you have an idea what could cause this?

**Comment by mbabker** [comment]
There haven't been any changes in the Registry API to cause this.

**Comment by Bakual** [comment]
@mbabker Found the culprit: https://github.com/joomla/joomla-cms/pull/12395
Apparently `registry->loadString()` worked also when you passed a registry object, however directly creating the registry object with `new Registry()` doesn't work with a registry object (doesn't make much sense anyway). It will become some funny Registry->data->data.

See https://github.com/joomla/joomla-cms/pull/12557

Closing as we have a PR.

**Comment by mbabker** [comment]
I guess that would make sense since `Registry` has a `__toString()` method so it could be typecast by something either in the formatter API or the underlying PHP engine.  Either way definitely makes no sense to take a registry, convert it to string, then convert it back.

**Comment by Bakual** [comment]
It may be worth to check if there are similar cases in other places of your merged PR.
I'll see if I can take the time tomorrow to look over it.

**Comment by brianteeman** [comment]
Great that you found the cause

**Comment by Bakual** [comment]
> It may be worth to check if there are similar cases in other places of your merged PR.

There were in fact some: https://github.com/joomla/joomla-cms/pull/12571
